src/CoreBundle/private/modules/CMSTemplateEngine/views/cmp_layoutlist.view.php

```diff
@@ -12,11 +25,20 @@
     <?php
     while ($oPageLayout = $data['oMasterDefs']->Next()) {
         /** @var $oPageLayout TdbCmsMasterPagedef */
-        $bIsActiveLayout = ($data['sActivePageDef'] == $oPageLayout->id); ?>
-        <li class="<?php if (($data['sActivePageDef']) == $oPageLayout->id) {
-            echo 'layoutitemactive';
-        } ?>"
-            onClick="parent.document.getElementById('userwebpageiframe').src='<?=URL_WEB_CONTROLLER; ?>?pagedef=<?=$data['id']; ?>&__masterPageDef=true&__modulechooser=true&id=<?=TGlobal::OutHTML(urlencode($oPageLayout->id)); ?>';">
+        $bIsActiveLayout = ($data['sActivePageDef'] == $oPageLayout->id);
+        $layoutItemClass = true === $bIsActiveLayout ? 'layoutitemactive' : '';
+
+        $urlParameters = [
+            'pagedef' => $data['id'],
+            '__masterPageDef' => 'true',
+            '__modulechooser' => 'true',
+            'id' => TGlobal::OutHTML($oPageLayout->id),
+            'previewLanguageId' => $languageService->getCmsBaseLanguageId(),
```

Comment:
preview language should be the edit language of the current user. Use the base language only as fallback if the current user has no edit language.